code/items/weapons/TTTWeapon.cs

```diff
@@ -28,7 +27,7 @@ public WeaponAttribute(string name) : base(name)
     }
 
     [WeaponAttribute("ttt_weapon")]
-    public abstract partial class Weapon : BaseWeapon
+    public abstract partial class TTTWeapon : BaseWeapon, IBuyableItem
     {
         public string Name { get; private set; }
```

Comment:
Can we get rid of this with `GetName()` now available?

Reply:
Good hint, I kept `name` as `private` instead saving performance instead of fetching the name everytime from a class attribute.